src/Microsoft.Data.Sqlite/SqliteCommand.cs

```diff
@@ -111,10 +112,10 @@ public new virtual SqliteDataReader ExecuteReader(CommandBehavior behavior)
             do
             {
                 Sqlite3StmtHandle stmt;
-                var rc = NativeMethods.sqlite3_prepare16_v2(
+                var rc = NativeMethods.sqlite3_prepare_v2(
                     Connection.DbHandle,
                     tail,
-                    -1,
+                    Encoding.UTF8.GetByteCount(tail) + 1, // add one to include the nul-terminator. See SQLite.org docs for sqlite3_prepare_v2.
```

Comment:
...haha. I was just thinking--is it faster for our managed code to find and use bytes.Length than for native code to find the null-terminator? I'm not sure. Maybe we need some real perf tests, not just our intuition.